numba/tests/test_np_functions.py

```diff
@@ -1078,6 +1078,42 @@ def check(a, v):
         bins = np.float64(list(bins) + [float('nan')] * 7) / 2.0
         values = np.arange(20) - 0.5
 
+        for a in (bins, list(bins)):
+            # Scalar values
+            for v in values:
+                check(a, v)
+            # Array values
+            for v in (values, values.reshape((4, 5))):
+                check(a, v)
+            # Sequence values
+            check(a, list(values))
+
+        # Third with unichr (no NaNs)
+        unicode_a = 97
+        alphabet_size = 26
+        string_length = 8
+
+        np.random.seed(42)
+        bins_unicode = np.random.choice(
+            np.arange(
+                unicode_a,
+                unicode_a + alphabet_size,
+                dtype=np.uint32,
+            ),
+            size=(5, string_length),
+        )
+        bins_unsorted = bins_unicode.view(f"<U{string_length}").reshape(-1)
+        bins = np.sort(bins_unsorted)
+        values_unicode = np.random.choice(
+            np.arange(
+                unicode_a,
+                unicode_a + alphabet_size,
+                dtype=np.uint32,
+            ),
+            size=(20, string_length),
+        )
+        values = values_unicode.view(f"<U{string_length}").reshape(-1)
```

Comment:
Not necessary for this PR but perhaps one should split the entire function into different test cases.